datafusion/sqllogictest/test_files/string/string.slt

```diff
@@ -48,13 +48,115 @@ from test_basic_operator;
 Andrew datafusion📊🔥 true false true false
 Xiangpeng datafusion数据融合 false true false true
 Raphael datafusionДатаФусион false false false false
+under_score un iść core false false false false
+percent pan Tadeusz ma iść w kąt false false false false
 NULL NULL NULL NULL NULL NULL
 
 #
 # common test for string-like functions and operators
 #
 include ./string_query.slt.part
 
+# TODO support all String types in sql_like_to_expr and move this test to `string_query.slt.part`
```

Comment:
Is this the same as https://github.com/apache/datafusion/issues/12637? 

If so perhaps we can add a reference to the ticket as a comment

If not, maybe we can create another issue -- is the issue it doesn't work in StringView? / Dictionary?